### DISCUSSION TITLE
Any risks with forwardRef?

### DISCUSSION THREAD

**Discussion by keithburgie-rts** [chakra-ui/chakra-ui#5102] [Q&A]
Is there a downside to wrapping a component in forwardRef? I'm maintaining a library that extends Chakra UI and keep adding forwardRef to components as needed. I'm considering just wrapping all containers, buttons, etc. with forwardRef in case of future need. How will I be shooting myself in the foot?

**Comment by Naoto-Ida** [comment]
Two years late, but while working on a design system (package) which wraps Chakra UI, we noticed that anything wrapped by forwardRef was not able to be tree shaken by NextJS.
